### PR TITLE
New Finnish Cheat Sheet

### PR DESCRIPTION
**Pull request by riqpe**
https://duck.co/ia/view/finnish_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@riqpe Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/finnish.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            "val": "Kaksikymmentäkaksi, kolmekymmentäkolme, neljäkymmentäneljä, ...",
+            "key": "twenty-two, thirty-three, forty-four, ..."
+        },{
+            "val": "Sata, kaksisataa, kolmesataa, ...",
+            "key": "Hundred, two hundred, three hundred, ..."
+        },{
+            "val": "Tuhat, kaksituhatta, kolmetuhatta, ...",
+            "key": "Thousand, two thousand, three thousand, ..."
+        },{
+            "val": "Viisimiljoonaa neljäsataakolmekymmentäyhdeksäntuhatta kaksisataakaksikymmentäyksi",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/finnish.json]: I think this one is going to cause some visual problems..

**Comment by moollaza** [comment]
@riqpe thanks for this! Are you a native Finnish speaker?

File: share/goodie/cheat_sheets/json/finnish.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
+{
+    "id": "finnish_cheat_sheet",
+    "name": "Finnish Cheat Sheet",
+    "description": "Basic Finnish words and phrases",
+    "aliases": ["finnish phrases", "english to finnish", "basic finnish phrases", "basic finnish"],
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/finnish.json]: Remove `finnish phrases` and `basic finnish phrases` (these are included for the template by default).

File: share/goodie/cheat_sheets/json/finnish.json (hunk 1 above)
Comment (riqpe) [inline share/goodie/cheat_sheets/json/finnish.json]: It seemed to be rendering just fine when I was testing. It nicely demonstrates how numbers are concatenated, but I guess I could remove it.

File: share/goodie/cheat_sheets/json/finnish.json [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    "template_type": "reference",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/finnish.json]: Lets set this to `language` so we get all the lovely benefits!

**Comment by riqpe** [comment]
@moollaza Yes I am. I saw the French cheat sheet a month back and thought I'd give a try. That's how I wounded up doing a few of these. :)

File: share/goodie/cheat_sheets/json/finnish.json (hunk 2 above)
Comment (riqpe) [inline share/goodie/cheat_sheets/json/finnish.json]: done.

File: share/goodie/cheat_sheets/json/finnish.json (hunk 3 above)
Comment (riqpe) [inline share/goodie/cheat_sheets/json/finnish.json]: Done. What are the benefits with `language` template_type?

**Comment by moollaza** [comment]
@riqpe Thanks! This LGTM :+1:

File: share/goodie/cheat_sheets/json/finnish.json (hunk 3 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/finnish.json]: @riqpe For one thing the triggering :wink:, it allows the cheat sheet to trigger on more language-related phrases. Additionally it provides access to the `trn` attribute in items.

**Pull request merged**